File: spring-rest-docs/springboot2-restdocs-example/src/main/java/com/ys/rest_docs_boot2/domain/member/MemberRepository.java
package com.ys.rest_docs_boot2.domain.member;

import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

public interface MemberRepository extends JpaRepository<Member, Long> {

    @Query("select (count(m) > 0) from Member m where m.nickName = ?1")
    boolean existsByNickName(
        String nickName);

    Optional<Member> findByNickName(String nickName);

}

File: graphql/graphqlex/web-1/src/main/kotlin/com/yscorp/web1/domain/MemberRepository.kt
package com.yscorp.web1.domain

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository

@Repository
interface MemberRepository : JpaRepository<Member, Long> {
    fun findByType(type: MemberType): List<Member>

    @Query(value = "SELECT m FROM Member m WHERE LOWER(m.firstName) like %:name%")
    fun fetchMembersByName(@Param("name") text: String): List<Member>
}
